src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1334,6 +1334,143 @@ def enable_sequential_cpu_offload(self, gpu_id: Optional[int] = None, device: Un
                 offload_buffers = len(model._parameters) > 0
                 cpu_offload(model, device, offload_buffers=offload_buffers)
 
+    def enable_group_offload(
+        self,
+        onload_device: torch.device,
+        offload_device: torch.device = torch.device("cpu"),
+        offload_type: str = "block_level",
+        num_blocks_per_group: Optional[int] = None,
+        non_blocking: bool = False,
+        use_stream: bool = False,
+        record_stream: bool = False,
+        low_cpu_mem_usage=False,
+        offload_to_disk_path: Optional[str] = None,
+        exclude_modules: Optional[Union[str, List[str]]] = None,
```

Comment:
I think it's okay to expose this as an argument as opposed to how we do `model_cpu_offload_seq`, for example:

https://github.com/huggingface/diffusers/blob/764b62473ac10afd9c52e6b3f3f528f719bc7a34/src/diffusers/pipelines/flux/pipeline_flux.py#L179

This is because model CPU offloading relies on a sequence for device management. I don't think we have that constraint in the case of group offloading.